code/modules/clothing/suits/cardborg_suit.dm

```diff
@@ -0,0 +1,261 @@
+	item_state = "cardborg_deathbot"
+	species_disguise = "High-tech killer robot"
+
+/*
+ / Disguise code.
+*/
+/obj/item/clothing/head/cardborg/equipped(mob/living/user, slot)	// Attempt to disguise when you put on the helmet.
+	..()
+	if(ishuman(user) && slot == SLOT_HUD_HEAD)
+		var/mob/living/carbon/human/H = user
+		if(istype(H.wear_suit, /obj/item/clothing/suit/cardborg))
+			var/obj/item/clothing/suit/cardborg/CB = H.wear_suit
+			CB.apply_borg_disguise(user, src)
+
+/obj/item/clothing/head/cardborg/dropped(mob/living/user)			// You stop being a robot if you remove your helmet.
+	..()
+	user.remove_alt_appearance("borg_disguise_variant")
+
+/obj/item/clothing/suit/cardborg/equipped(mob/living/user, slot)	// Attempt to disguise when you put on the suit.
+	..()
+	if(slot == SLOT_HUD_OUTER_SUIT)
+		apply_borg_disguise(user)
+
+/obj/item/clothing/suit/cardborg/dropped(mob/living/user)			// You stop being a robot if you remove your suit.
+	..()
+	user.remove_alt_appearance("borg_disguise_variant")
+
+/obj/item/clothing/suit/cardborg/proc/apply_borg_disguise(mob/living/carbon/human/H, obj/item/clothing/head/cardborg/borghead)
+	if(!istype(H))
+		return
+	if(!borghead)
+		borghead = H.head									// This actually stops the disguise from applying just from having it in your hands AGHHHH!
+	if(istype(borghead, /obj/item/clothing/head/cardborg))	// Why is this done this way? because equipped() is called BEFORE THE ITEM IS IN THE SLOT WHYYYY!
+		var/borg_disguise_variant
+		var/disguise_eyes
+		borg_disguise_variant = pick(borghead.available_disguises)
+		switch(borg_disguise_variant)	// We need to know what glowy bits to stick on.
+			if("Standard", "Standard-Secy", "Standard-Engi", "Standard-Mine", "Standard-Serv", "Standard-Jani")
+				disguise_eyes = "eyes-Standard"
+			if("Cricket-SEC", "Cricket-ENGI", "Cricket-MINE", "Cricket-SERV", "Cricket-MEDI", "Cricket-Jani")
+				disguise_eyes = "eyes-Cricket"
+			if("ertgamma", "nano_bloodhound")
+				disguise_eyes = "eyes-ertgamma"
+			if("bloodhound")
+				disguise_eyes = "eyes-bloodhound"
+			if("syndie_bloodhound")
+				disguise_eyes = "eyes-syndie_bloodhound"
+			if("syndi-engi")
+				disguise_eyes = "eyes-syndi-engi"
+			if("lavaland")
+				disguise_eyes = "eyes-lavaland"
+			if("Miner")
+				disguise_eyes = "eyes-Miner"
+			if("droid-miner")
+				disguise_eyes = "eyes-droid-miner"
+			if("droid-medical")
+				disguise_eyes = "eyes-droid-medical"
+			if("landmate")
+				disguise_eyes = "eyes-landmate"
+			if("Engineering")
+				disguise_eyes = "eyes-Engineering"
+			if("surgeon")
+				disguise_eyes = "eyes-surgeon"
+			if("mopgearrex")
+				disguise_eyes = "eyes-mopgearrex"
+			if("toiletbot")
+				disguise_eyes = "eyes-toiletbot"
+			if("qualified_doctor")
+				disguise_eyes = "eyes-qualified_doctor"
+			if("custodiborg")
+				disguise_eyes = "eyes-custodiborg"
+			if("heavySec")
+				disguise_eyes = "eyes-heavySec"
+			if("squatminer")
+				disguise_eyes = "eyes-squatminer"
```

Comment:
Can you make this into an associated list? this shouldn't need the type that contra showed you, but instead my simpler variant should work better here